### PR TITLE
[WIP] Initial attempt at replacing nuget pack with Paket pack

### PR DESCRIPTION
**Pull request by Vidarls**
Having some strange issue with Paket.Pack, so I have not yet verified that this code will work.

Pushing it for troubleshooting purposes on fsprojects/Paket#645 and early feedback.

Currently missing: 

- more details in template file.

### CODE REVIEW

**Comment by forki** [comment]
https://github.com/fsprojects/Paket/commit/b09e364de56b828dd6ed3ac1394412bb54056130

please give version via cmd line param (FAKE task) and add description via template file.

File: build.template [hunk 1]
```diff
@@ -171,23 +171,20 @@ Target "SourceLink" (fun _ ->
...
-            AccessKey = getBuildParamOrDefault "nugetkey" ""
-            Publish = hasBuildParam "nugetkey"
-            Dependencies = [] })
-        ("nuget/" + project + ".nuspec")
+            Version = release.NugetVersion
+            ReleaseNotes = toLines release.Notes})
+)
+
+Target "PublishNuget" (fun _ ->
+    Paket.Pack(fun p ->
```
Comment (forki) [inline build.template]: this should be Push

**Comment by Vidarls** [comment]
Latest Paket + version in cmdline + description in template file still fails :disappointed:

**Comment by forki** [comment]
I wish I could reproduce...

File: src/FSharp.ProjectTemplate/paket.template [hunk 2]
```diff
@@ -0,0 +1,6 @@
+type project
+authors 
+    ##Author##
+description
+    ##Description##
+
```
Comment (forki) [inline src/FSharp.ProjectTemplate/paket.template]: the ID is still missing

**Comment by forki** [comment]
https://github.com/fsprojects/ProjectScaffold/pull/143/files#r25256295

File: src/FSharp.ProjectTemplate/paket.template (hunk 2 above)
Comment (forki) [inline src/FSharp.ProjectTemplate/paket.template]: but wait. I can load the Id from the assembly without custom attributes. give me a minute

Comment (forki) [inline src/FSharp.ProjectTemplate/paket.template]: https://github.com/fsprojects/Paket/commit/83f0831a3650e1d411acc5559822d753bc59779d
I press all my thumbs.

**Comment by Vidarls** [comment]
Oh yeah 

celebration

Now it is working.

**Comment by forki** [comment]
just tell me when you think we can merge this or about any other issue.

**Comment by Vidarls** [comment]
I will review it, clean it up and add some more stuff to the template file in the morning. Will let you know when it is ready for review and merge

**Comment by Vidarls** [comment]
Should be ready for review / merge now.

Not sure why Travis is failing, looks like Paket failed to download something:

```
Paket failed with:
	Could not retrieve data from https://api.github.com/repos/fsharp/FAKE/commits/master

Message: The remote server returned an error: (403) Forbidden.
```

File: init.fsx [hunk 3]
```diff
@@ -143,6 +141,12 @@ let replaceContent file =
   |> replace (oldProjectGuid.ToUpperInvariant()) (projectGuid.ToUpperInvariant())
   |> replace (oldTestProjectGuid.ToUpperInvariant()) (testProjectGuid.ToUpperInvariant())
   |> replace solutionTemplateName projectName
+  |> replaceWithVarOrMsg "##Author##" "Author Not Set" 
+  |> replaceWithVarOrMsg "##Description##" "Description Not Set"
```
Comment (forki) [inline init.fsx]: better "Description not set" ?

File: init.fsx [hunk 4]
```diff
@@ -143,6 +141,12 @@ let replaceContent file =
   |> replace (oldProjectGuid.ToUpperInvariant()) (projectGuid.ToUpperInvariant())
   |> replace (oldTestProjectGuid.ToUpperInvariant()) (testProjectGuid.ToUpperInvariant())
   |> replace solutionTemplateName projectName
+  |> replaceWithVarOrMsg "##Author##" "Author Not Set" 
+  |> replaceWithVarOrMsg "##Description##" "Description Not Set" 
+  |> replaceWithVarOrMsg "##Summary##" "Project has no summmaryrt"
```
Comment (forki) [inline init.fsx]: typo

File: init.fsx [hunk 5]
```diff
@@ -143,6 +141,12 @@ let replaceContent file =
...
+  |> replaceWithVarOrMsg "##Author##" "Author missing"
```
Comment (forki) [inline init.fsx]: I think this one is a duplicate

File: src/FSharp.ProjectTemplate/paket.template [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+owners
+    ##Author##
+authors 
+    ##Author##
+projectUrl
+    http://github.com/##GitHome##/FSharp.ProjectScaffold
+iconUrl
+    https://raw.githubusercontent.com/fsprojects/ProjectScaffold/master/docs/files/img/logo.png
+licenseUrl
+    http://github.com/##GitHome##/FSharp.ProjectScaffold/blob/master/LICENSE.txt
```
Comment (forki) [inline src/FSharp.ProjectTemplate/paket.template]: I think we need to replace FSharp.ProjectScaffold here

**Comment by forki** [comment]
Good work! only couple of minor comments. The travis error seems related to github API limit.

dance

File: src/FSharp.ProjectTemplate/paket.template (hunk 6 above)
Comment (Vidarls) [inline src/FSharp.ProjectTemplate/paket.template]: Fsharp.Projectscaffold is being replaced.

**Comment by Vidarls** [comment]
Fixed typos and duplicates, should be ready now.

**Comment by forki** [comment]
cool. @pblasucci what do you think? Do we want to switch over to paket for nuget pack and push?

**Comment by pblasucci** [comment]
This looks terrific. Well done!

**Comment by forki** [comment]
https://github.com/fsprojects/ProjectScaffold/commit/186536797e6f4241994bc7eb11e691a2fdef46d8

**Pull request merged**